kubernetes/scenescape-chart/templates/pgserver-dep.yaml

```diff
@@ -21,145 +21,81 @@ spec:
-      - name: chown-initcontainer
-        image: busybox
-        command: ["/bin/sh", "-c"]
-        args:
-        - |
-          if [ -e /workspace/db/PG_VERSION ]; then
-            echo "Previous database exists. Not running chown."
-            exit 0
-          else
-            echo "Previous database doesn't exist. Running chown command on /workspace."
-            chown -R 1000:1000 /workspace
-          fi
-        volumeMounts:
-        - mountPath: /workspace
-          name: pgserver-storage
-          subPath: workspace
-        - mountPath: /workspace/media
-          name: media-storage
+        - name: chown-initcontainer
+          image: busybox
+          volumeMounts:
+            - mountPath: /workspace
+              name: pgserver-storage
+              subPath: workspace
       containers:
-        - args:
-          - database
-          - --preloadexample
-          image: {{ .Values.repository }}/{{ .Values.pgserver.image }}:{{ .Chart.AppVersion }}
-          name: {{ .Release.Name }}-pgserver
+        - name: {{ .Release.Name }}-pgserver
+          image: {{ if .Values.pgserver.repository }}{{ .Values.pgserver.repository }}/{{ end }}{{ .Values.pgserver.image }}
           env:
-          - name: SUPASS
-            value: {{ required "You must set supass (e.g. --set supass=...) for this chart to install." .Values.supass | quote }}
-{{- if ne (default .Values.dbroot "") "" }}
-          - name: DBROOT
-            value: {{ .Values.dbroot }}
-{{- end }}
+            - name: POSTGRES_USER
+              value: "scenescape"
+            - name: POSTGRES_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Release.Name }}-db
+                  key: DATABASE_PASSWORD
+            - name: POSTGRES_DB
+              value: "scenescape"
 {{- if .Values.proxy.enabled }}
-          - name: HTTP_PROXY
-            value: {{ .Values.proxy.httpProxy }}
-          - name: HTTPS_PROXY
-            value: {{ .Values.proxy.httpsProxy }}
-          - name: NO_PROXY
-            value: {{ .Values.proxy.noProxy }}
-          - name: http_proxy
-            value: {{ .Values.proxy.httpProxy }}
-          - name: https_proxy
-            value: {{ .Values.proxy.httpsProxy }}
-          - name: no_proxy
-            value: {{ .Values.proxy.noProxy }}
+{{- with .Values.proxy }}
+            - name: HTTP_PROXY
+              value: {{ .httpProxy }}
+            - name: HTTPS_PROXY
+              value: {{ .httpsProxy }}
+            - name: NO_PROXY
+              value: {{ .noProxy }}
+            - name: http_proxy
+              value: {{ .httpProxy }}
+            - name: https_proxy
+              value: {{ .httpsProxy }}
+            - name: no_proxy
+              value: {{ .noProxy }}
+{{- end }}
 {{- end }}
           imagePullPolicy: Always
           securityContext:
             privileged: true
             capabilities:
               add: ["SYS_ADMIN"]
           ports:
```

Comment:
The privileged: true and SYS_ADMIN capability are not needed for the database service. Granting these privileges increases security risk without providing any benefit for DB operations. We should remove them to follow the principle of least privilege. Most likely this was added to be able to mount `/dev/fuse`.